client.go

```diff
@@ -151,7 +165,26 @@ func (c *Client) listen() {
 					}
 				}
 
-				app.forward(c.id, c.session, boot)
+				// TODO: Ugly. Find a better way to wrap m.data into {"args": m.data}.
+				body := map[string]map[string]string{}
+				argsMap := map[string]string{}
+				json.Unmarshal(boot, &argsMap)
+				body["args"] = argsMap
+
+				if getRequestHeaders, ok := c.broker.site.clientIDToHeaders[c.id]; ok {
+					headersMap := map[string]string{}
+					mergeHeaders(headersMap, *getRequestHeaders, *c.header)
+					body["headers"] = headersMap
+					// Send headers just once as they are stored in Wave app anyway.
+					delete(c.broker.site.clientIDToHeaders, c.id)
```

Comment:
marshaling below can fail. Shouldn't we remove the headers only after we are sure that the operation succeeded?

Reply:
Outdated now, headers are now stored on WS client instance which means they are automatically garbage collected when the connection terminates.